Lib/fontTools/varLib/instancer.py

```diff
@@ -1099,6 +1117,293 @@ def pruningUnusedNames(varfont):
             del varfont["ltag"]
 
 
+def updateNameTable(varfont, axisLimits):
+    """Update an instatiated variable font's name table using the Axis
+    Values from the STAT table.
+
+    The updated nametable will conform to the R/I/B/BI naming model.
+    """
+    if "STAT" not in varfont:
+        raise ValueError("Cannot update name table since there is no STAT table.")
+    stat = varfont["STAT"].table
+    if not stat.AxisValueArray:
+        raise ValueError("Cannot update name table since there are no STAT Axis Values")
+    fvar = varfont["fvar"]
+
+    # The updated name table must reflect the new 'zero origin' of the font.
+    # If we're instantiating a partial font, we will populate the unpinned
+    # axes with their default axis values.
+    fvarDefaults = {a.axisTag: a.defaultValue for a in fvar.axes}
+    axisCoords = deepcopy(axisLimits)
+    for axisTag, val in fvarDefaults.items():
+        if axisTag not in axisCoords or isinstance(axisCoords[axisTag], tuple):
+            axisCoords[axisTag] = val
+
+    # To get the required Axis Values for the zero origin, we can simply
+    # duplicate the STAT table and instantiate it using the axis coords we
+    # created in the previous step.
+    stat_new = deepcopy(stat)
+    _instantiateSTAT(stat_new, axisCoords)
```

Comment:
It's not entirely clear to me why we have to call instantiateSTAT on a copy of the actual table here. Could we avoid it by running the updateNameTable after the normal instantiateSTAT has run?

Reply:
To construct the name table records, we need a single AxisValue each axis. If the user instantiates a partial font, there could potentially be AxisValues for a single axis. By reinstantiating a duplicate STAT table using the new axisCoords, we will get a single AxisValue for each axis.

I haven't looked at this code in a while and this took me a fair bit of time to work out. I may refactor this.